### PR TITLE
Initial support for a master survey

### PR DESCRIPTION
**Pull request by drewpotter**
Here is the initial support for a master survey through the use of a boolean and the various links to take the survey having been implemented across Go-Fox.

### CODE REVIEW

File: app/models/survey.rb [hunk 1]
```diff
@@ -46,4 +47,8 @@ def questions
 
     Question.order("RANDOM()") + demographic_questions + customised_questions
   end
+
+  def self.master_survey
+    where(master: true).first
```
Comment (spencerldixon) [inline app/models/survey.rb]: I think we could use `find_by_master(true)` here which will return the first result. Also, we should probably put some kind of validation or contstraint that means you can only have one master survey at a time.

Comment (drewpotter) [reply app/models/survey.rb]: Thanks - would you recommend using a before_save kind of action to achieve that constraint?

Comment (spencerldixon) [reply app/models/survey.rb]: Probably just a uniqueness validation https://stackoverflow.com/questions/19575336/rails-custom-validation-only-one-record-can-be-true

Comment (drewpotter) [reply app/models/survey.rb]: Many thanks I am working to implement this.

File: app/models/survey.rb [hunk 2]
```diff
@@ -46,4 +51,22 @@ def questions
...
+    return unless master?
+
+    matches = Survey.master
+    if persisted?
+      matches = matches.where('id != ?', id)
+    end
+    if matches.exists?
+      errors.add(:master, "Can only have one master survey")
+    end
+  end
```
Comment (spencerldixon) [inline app/models/survey.rb]: @drewpotter if you read a little further down on the stack overflow post there is a much more elegant solution

**Comment by benkt2** [comment]
Couple of comments: 
after db:migrate and db:seed, I don't have a master survey in the database and the front page 500s - if I UPDATE the db directly to set one of the existing surveys to master then front page works fine.
If I then try to create another master survey (once logged in) it doesn't work but I get a very ugly 500 page which doesn't explain the problem.
If I have a master survey that is unpublished, front page loads but link to take the survey gives "You cannot submit a response for an unpublished survey" - which at least makes sense.

What's confusing me is - what's the point of the master survey option? If a master survey has to exist (and be published) for the site to work, why is there an offer to set the flag on new surveys?

**Comment by drewpotter** [comment]
I'm sorry that you experienced that @benkt2. I am not sure on the best way of setting an initial master survey, but perhaps the db:seed should indeed take care of that.  The 500 page when you try to set another as master wasn't happening to me locally. I was getting a message stating that there is already a master survey. The offer to set it, is so that essentially the admins of Go-Fox would be able to set any survey as the master. However I think I will rework how this works and change a few things. Thanks for the feedback.

**Pull request merged**